src/templates/nulldata.js

```diff
@@ -14,15 +14,23 @@ function check (script) {
 check.toJSON = function () { return 'null data output' }
 
 function encode (data) {
-  typeforce(types.Buffer, data)
+  typeforce(typeforce.oneOf(types.Buffer, types.Array), data)
 
-  return bscript.compile([OPS.OP_RETURN, data])
+  if (!types.Array(data)) {
+    data = [ data ]
+  }
+
+  return bscript.compile([OPS.OP_RETURN].concat(data))
 }
 
 function decode (buffer) {
   typeforce(check, buffer)
 
-  return buffer.slice(2)
+  var chunks = bscript.decompile(buffer)
+
+  chunks.shift()
+
+  return chunks.length === 1 ? chunks[0] : chunks
```

Comment:
I suggest this might worth clarifying if we're thinking about a break. Although the majority of `OP_RETURN` usage is just one push, they can also have zero or several data-pushes, it might be good to return an array here. The content after OP_RETURN hardly gets executed, does it even have to be valid script?

Whatever templates we have, it's clear that people expect them not to conditionally fail (suppose the function couldn't recognize `OP_RETURN [DATA] [DATA]` and classified it non-standard) and that whatever we call 'nullData' doesn't change in the future..

I noticed we don't check the first opcode == OP_RETURN - this is another degree of freedom to consider when we call this script 'nullData'..

Maybe we should outline what exactly we consider a nullData script.. Thoughts?